databricks/koala/structures.py

```diff
@@ -511,6 +511,29 @@ def head(self, n=5):
         df._metadata = self._metadata.copy()
         return df
 
+    def rename(self, mapper=None, index=None, columns=None, axis=None):
+        if axis is not None and (index is not None or columns is not None):
```

Comment:
Did you get inspired from a specific source of code to implement this logic? If this is the case, it would be nice to mention it here.

Reply:
Not per se - I just tried the corner cases mentioned in pandas docs and implemented it here trying to match the same error message and error type